app/src/main/java/com/example/tungcon/demo/movie/MovieAdapter.java

```diff
@@ -0,0 +1,94 @@
+    private List<Movie> mMovieList;
+    OnImageClickListener mOnImageClickListener;
+
+    public MovieAdapter(Context mContext, List<Movie> mMovieList,
+                        OnImageClickListener mOnImageClickListener) {
+        this.mContext = mContext;
+        this.mMovieList = mMovieList;
+        this.mOnImageClickListener = mOnImageClickListener;
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        LayoutInflater inflater = LayoutInflater.from(mContext);
+        View view = inflater.inflate(R.layout.item_movie, parent, false);
+        return new ViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
+        holder.bindData(mMovieList.get(position));
+    }
+
+    @Override
+    public int getItemCount() {
+        return mMovieList.size();
+    }
+
+    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+        private TextView mTextViewMovie, mTextViewYear, mTextViewDuration;
+        private TextView mTextViewDirector, mTextViewTagline, mTextViewCast, mTextViewStory;
+        private ImageView mImageMovie;
+        private RatingBar mRatingMovie;
+        private CardView mCardView;
+
+        public ViewHolder(View itemView) {
+            super(itemView);
+            mTextViewYear = itemView.findViewById(R.id.text_year);
+            mTextViewDuration = itemView.findViewById(R.id.text_duration);
+            mTextViewStory = itemView.findViewById(R.id.text_story);
+            mTextViewTagline = itemView.findViewById(R.id.text_tagline);
+            mTextViewMovie = itemView.findViewById(R.id.text_movie);
+            mTextViewDirector = itemView.findViewById(R.id.text_driction);
+            mTextViewCast = itemView.findViewById(R.id.text_cast);
+            mRatingMovie = itemView.findViewById(R.id.rating_movie);
+            mImageMovie = itemView.findViewById(R.id.image_movie);
+            mCardView = itemView.findViewById(R.id.layout_item);
+
+        }
+
+        public void bindData(Movie movie) {
+            mTextViewMovie.setText(movie.getMovie());
+            mTextViewYear.setText(movie.getYear() + "");
+            mRatingMovie.setRating(movie.getRating());
+            mTextViewTagline.setText(movie.getTagline());
+            mTextViewCast.setText(movie.getCastList());
+            mTextViewDirector.setText(movie.getDirDector());
+            mTextViewStory.setText(movie.getStory());
+            mTextViewDuration.setText(movie.getDuration());
+            Glide.with(mContext).load(movie.getImage()).into(mImageMovie);
+            mCardView.setOnClickListener(this);
+        }
+
+        @Override
+        public void onClick(View view) {
+            switch (itemView.getId()) {
+                case R.id.layout_item:
+                    mOnImageClickListener.onClick(mMovieList.get(getAdapterPosition()));
```

Comment:
```suggestion
                    if (mOnImageClickListener == null) {
                        return;
                    }
                    mOnImageClickListener.onClick(mMovieList.get(getAdapterPosition()));
```

check null listenner